### ISSUE TITLE
[SB2] Difficult to awake from sleep

### ISSUE THREAD

**Issue by junalmeida** [linux-surface/linux-surface#1624]
[Description of the bug or feature]

After going to sleep, I have to press the power button two to three times to see the wayland lockscreen appear on the monitor. 

Each time I press it, I hear a sound like usb connecting, but no image on external monitor. 

I use the laptop closed, showing only on external monitor, which is a 5120x1440 monitor on a usb-c->dp cable. 

### Environment

- Hardware model: Surface Book 2 (with up-to-date firmware)
- Kernel version: Linux marcos-surfacebook2 6.10.10-surface-1 #1 SMP PREEMPT_DYNAMIC Sun Sep 15 03:41:08 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
- Distribution: KDE Neon (Ubuntu Noble)

[dmesg.txt](https://github.com/user-attachments/files/17960276/dmesg.txt)

**Comment by slopsjon** [comment]
Similar problem with my SP7. I have to tap the space bar in the password box to stop it going back into sleep. Only happened in the last day or so.

Linux fedora 6.11.4-1.surface.fc41.x86_64 #1 SMP PREEMPT_DYNAMIC Sun Oct 20 18:44:04 UTC 2024 x86_64 GNU/Linux
Cinnamon 6.2.9

**Comment by junalmeida** [comment]
@slopsjon  Sorry not what I meant to say. I don't have any issue on it going to sleep. It goes to sleep when it is supposed to. I have an issue when I need to wake it up from sleep.

**Comment by slopsjon** [comment]
Yes that's what happening to me. It goes to sleep ok, when I press the power button to wake it, It shows the log on screen then goes back to sleep. I have to bash at the keyboard to try an interupt it, to stop it going back to sleep a second later.

**Comment by systemofapwne** [comment]
Same issue here on an SP7 with Arch.
Not sure, if it is caused by changes to this kernel or an update to my DE (KDE), that somehow sees the pressed button on wake again and interprets it as "go to sleep now".
This was not alwas like this.

**Edit:**
By unbinding the power button from "sleep-mode" in my DE, I can now successfully start my SP7 again without having it immediately go to sleep again.

I rebound it to the logout-screen now to have at least a bit of convenience now to quickly go to sleep via the power button.
![Image](https://github.com/user-attachments/assets/29b37ae1-cccc-4801-abfc-86001fcc21cf)
When I now log back in, the logout screen appears for a short amount of time, further giving support to my hypothesis that the "wake" button press is again intepreted immediately after wake.

Even though it looks like changes to the KDE have introduced this issue, it still could be the kernel, somehow still propagating the button press of the wake event into the running session.

**Comment by Cal-Nine** [comment]
Same issue for me on SP7, screen flashes on when waking using power button, then immediately flashes off. Same fix works for me in Plasma.
